### DISCUSSION TITLE
The null check generated in version 1.4.2 also satisfies the empty string check?

### DISCUSSION THREAD

**Discussion by GaoShuCheng** [mapstruct/mapstruct#2896] [General]
@Override
    public MyPojo toMyPojo(GeneratedPojo generatedPojo) {
        if ( generatedPojo == null ) {
            return null;
        }

        MyPojo myPojo = new MyPojo();

        try {
            if ( generatedPojo.getTargetTime() != null ) {
                myPojo.setSourceTime( new SimpleDateFormat( "yyyy-MM-dd HH:mm:ss" ).parse( generatedPojo.getTargetTime() ) );
            }
        }
        catch ( ParseException e ) {
            throw new RuntimeException( e );
        }

        return myPojo;
    }

If the property of type time is only checked for NULL, the time is formatted as an empty string is throws exception

**Comment by filiphr** [comment]
If you want to provide a custom presence check method you can use the new `@Condition` from 1.5. MapStruct will then use that to check whether it should map the target time. 

e.g.
```java
@Mapper
public interface MyMapper {

    MyPojo toMyPojo(GeneratedPojo pojo);

    @Condition
    default boolean isNotEmpty(String value) {
        return value != null && !value.isEmpty();
    }

}

```


Keep in mind that if your target time has incorrect format there will be an exception as well, ignoring empty strings is not the right approach for this.

**Reply by GaoShuCheng** [reply]
Thank you for your answer,I got it.
